### PR TITLE
expression, executor, plan: rewrite builtin function hex

### PR DESCRIPTION
**Pull request by SteveZhangBit**

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@XuHuaiyu @zz-jason PTAL

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1002,39 +1003,57 @@ type hexFunctionClass struct {
...
 func (c *hexFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinHexSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	switch t := args[0].GetTypeClass(); t {
+	case types.ClassString:
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		// Use UTF-8 as default
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: what does this comment mean?
why set `flen = arg0.Flen*3*2` ?

Comment (SteveZhangBit) [inline expression/builtin_string.go]: hex(str) will return the hex string of the original string byte by byte, so for utf-8 string, it should be *3 for each rune and *2 for each byte.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**